forms/collections/pages.js

```diff
@@ -35,12 +41,6 @@ define(function (require) {
       view.render();
       currentform.attributes._view.$el.append(view.el);
 
-      this.forEach(function (page, number) {
-        if (number !== index) {
-          page.removeView();
-        }
-      });
-
```

Comment:
> its this loop that is causing double rendering, doesn't matter where in the function it is.

I think it's the absence of any call to `.removeView()` for the current page that is causing the double rendering, which you address further up with the new lines. I agree that the current loop as-is does not help, but it isn't the cause.

What I was suggesting is basically removing the `if`-statement from the loop so all page views are always removed, including the current one.

But, as you say, the loop itself probably wasn't 100% necessary in the first place, and just catches theoretical edge cases.